### ISSUE TITLE
Incorrect d$FN vector in JM_BR.R

### ISSUE THREAD

**Issue by vshekar** [LanceFiondella/srt.core#29]
In the JM model specifically [this function](https://github.com/LanceFiondella/srt.core/blob/b265ec37bc5c1ff86bb4987ea945077b52eeb186/models/JM/JM_BM.R#L216) the d$FN vector seems to be incorrect. For example, dataset SYS1 has 136 points which means d$FN should start from 137 but starts from 136 which is not right.

A [corresponding function in GO model](https://github.com/LanceFiondella/srt.core/blob/b265ec37bc5c1ff86bb4987ea945077b52eeb186/models/GO/GO_BM_FT.R#L127) also uses the same variable and correctly starts from 137 for SYS1.

This has lead to the incorrect calculation of the number of predictions the model can make. JM makes 1 more prediction than it's supposed to.

**Comment by vshekar** [comment]
Fixed the issue. But not sure if correct. Changed the floor function to round here :  [RunModels.R](https://github.com/LanceFiondella/srt.core/blob/792d88d6694579bee356f7ec8e8bd71988dd1bd9/utility/RunModels.R#L109).

Can someone explain to me or add comments as to what the if else statement does? The else block is entered only for the JM model and incorrectly calculates the lower_pred_bound to be 136 rather than 137 when the floor function is used.

I'll close the bug once this is explained.

**Comment by arhik** [comment]
I didn't quite get the part from the first comment

> n the JM model specifically this function the d$FN vector seems to be incorrect. For example, dataset SYS1 has 136 points which means d$FN should start from 137 but starts from 136 which is not right.

**Comment by vshekar** [comment]
Sure, I'll try to explain. d$FN exists in both JM and GO models, an expectation would be that vector is be the same for both models. What I gathered from examining the vector is that it contains the point numbers to be predicted. SYS1 has 136 points, and suppose I want to predict failure times of 8 points after that. 


This means d$FN should be [137, 138, 139, ..., 144] for both models. But I found that in the JM model d$FN = [136, 137,..., 143]. This error was cascading into other functions which lead to JM predicting the wrong number of points. Hope that makes sense. 

----- Original Message -----

**Comment by arhik** [comment]
MVF_Inv function is just the inverse of the data points it receives. So any extra data point or missing data point is purely mistake else where data comes from(And I suspect its run models mishandling the data points). The functions are fine. And I want it to be purely mathematical from 'Model contribution' point of view. So any changes to it will confuse contributors. Also data returned from this function should be handled elsewhere as needed because this is a mathematical modeling file. I hope I am making sense.
>This means d$FN should be [137, 138, 139, ..., 144] for both models. But I found that in the JM model d$FN = [136, 137,..., 143]. This error was cascading into other functions which lead to JM predicting the wrong number of points. Hope that makes sense.

The missing data point is clearly the data handling issue prior to calling the `MVF_Inv` function. The extra logic in the `MVF_Inv` is a bit scary from contributors perspective.

**Comment by arhik** [comment]
Let me try with just the rounding part in runModels. I will let you know if any thing is abnormal.

**Comment by vshekar** [comment]
You are right, RunModels is mishandling the data which was what my 2nd comment in this thread talks about. The extra logic to replace Nan values was to keep everything else from breaking, which I had added before raising this issue: https://github.com/LanceFiondella/srt.core/issues/28. In the issue I mentioned that we can remove the logic once I figure out what RunModels does.

**Comment by arhik** [comment]
Regarding #28: Sorry I was busy then.

**Comment by arhik** [comment]
Shekar can you point me to the bug you fixed by using `max` to avoid `nan` values.

**Comment by vshekar** [comment]
There was no specific issue raised for it as far as I know. But if you asked the model to predict more points than it can and tried to plot it, it would show an error on the browser. You can probably recreate it by removing the max logic and set the finite model param to False.
